scio-elasticsearch/es7/src/main/scala/com/spotify/scio/elasticsearch/IndexAdmin.scala

```diff
@@ -78,4 +89,122 @@ object IndexAdmin {
       new CreateIndexRequest(index).source(mappingSource, XContentType.JSON),
       RequestOptions.DEFAULT
     )
+
+  /**
+   * Delete index
+   * @param index to be deleted
+   * @param timeout defaults to 1 minute
+   * @return Failure or unacknowledged response if operation did not succeed
+   */
+  def removeIndex(
+    nodes: Iterable[HttpHost],
+    index: String,
+    timeout: TimeValue = TimeValue.timeValueMinutes(1)
+  ): Try[AcknowledgedResponse] = {
+    val esOptions = ElasticsearchOptions(nodes.toSeq)
+    removeIndex(esOptions, index, timeout)
+  }
+
+  /**
+   * Delete index
+   * @param index to be deleted
+   * @param timeout defaults to 1 minute
+   * @return Failure or unacknowledged response if operation did not succeed
+   */
+  def removeIndex(
+    esOptions: ElasticsearchOptions,
+    index: String,
+    timeout: TimeValue
+  ): Try[AcknowledgedResponse] =
+    indicesClient(esOptions)(client => removeIndex(esOptions, index, client, timeout))
+
+  /**
+   * Delete index
+   * @param index to be deleted
+   * @param timeout defaults to 1 minute
+   * @return Failure or unacknowledged response if operation did not succeed
+   */
+  private def removeIndex(
+    esOptions: ElasticsearchOptions,
+    index: String,
+    client: IndicesClient,
+    timeout: TimeValue
+  ): AcknowledgedResponse = {
+    val request = new DeleteIndexRequest(index)
+      .timeout(timeout)
+
+    client.delete(request, RequestOptions.DEFAULT)
+  }
+
+  /**
+   * Add index alias and remove the alias from all other indexes if it is already pointed to any.
+   * If index already exists or some other error occurs this results in a [[scala.util.Failure]].
+   *
+   * @param alias        to be re-assigned
+   * @param newIndexName index to point the alias to
+   * @param timeout defaults to 1 minute
+   */
+  def swapIndexAlias(
```

Comment:
I think we can make this a little bit more broader.

What about renaming this to `createAlias` and have a `removePrevious: Boolean` and a `isWriteIndex: Boolean`. I think it would allow more use cases.

Reply:
Oops, missed this comment, sorry @regadas. To make sure I understood you right:
```createAlias``` would just add a new index alias
```removePrevious``` would unlink a given alias from all indices

What does ```isWriteIndex``` do?
I assume the Boolean would be derived from ```Try(AcknowledgedResponse)```  by applying ```x.map(_.isAchnowledged)getOrElse(false)```

Would be kinda great to still add a swap method since that's what we do when we create index from scratch every time

Reply:
Hi @6c697a61, ES alias can point to multiple indices and `isWriteIndex` will tell that a given index is the one to write to when using the alias to write. (It wouldn't actually be a `Boolean` in this case).

I think swapping can be achieved by using smth like `createAlias(alias, index, removePrevious = true)` without having specific `swap` op.

Reply:
Ohh, have them as additional parameters! Got confused, sorry - thought you meant functions. 
Adding ```removePrevious``` , but still not clear of how  ```isWriteIndex``` is intended to be used. I understand what it is, just want to make sure we are on the same page about how it should be used. Do you mean we could put it as an additional parameter in ensureIndex to be able to prevent writing to an aliasedIndex @regadas?

Reply:
Also, would it make sense to also add a way to obtain ElasticserchOptions from address string?
```  
def elasticSearchOptions(esAddress: String): ElasticsearchOptions = 
    ElasticsearchOptions(DnsResolver.addresses(esAddress))
  
```